relay-server/src/services/health_check.rs

```diff
@@ -182,127 +198,73 @@ impl HealthCheckService {
         }
     }
 
-    async fn handle_is_healthy(&self, message: IsHealthy) -> Status {
-        let upstream = self.upstream_relay.clone();
-
+    async fn check_readiness(&mut self, shutdown: &ShutdownHandle) -> Status {
+        // Internal metric that we need to be logged in recurring intervals. This is a form of
+        // health check, but it does not contribute to the status of this service.
         if self.config.relay_mode() == RelayMode::Managed {
-            let fut = upstream.send(IsNetworkOutage);
-            tokio::spawn(async move {
-                if let Ok(is_outage) = fut.await {
-                    metric!(gauge(RelayGauges::NetworkOutage) = u64::from(is_outage));
-                }
-            });
-        }
-
-        if matches!(message, IsHealthy::Liveness) {
-            return Status::Healthy;
+            if let Ok(is_outage) = self.upstream_relay.send(IsNetworkOutage).await {
+                metric!(gauge(RelayGauges::NetworkOutage) = u64::from(is_outage));
+            }
         }
 
-        if self.is_shutting_down.load(Ordering::Relaxed) {
+        if shutdown.get().is_some() {
             return Status::Unhealthy;
         }
 
-        let (sys_mem, auth, agg, proj) = tokio::join!(
-            self.probe("system memory", self.system_memory_probe()),
+        // TODO(ja): log like probe
+        let sys_mem = self.system_memory_probe();
```

Comment:
I'll address this TODO after an initial round of feedback. 

The issue here is that the system memory probe requires mutable access to the `System` instance, which means we cannot join it with other futures. However, since it is entirely sync we don't need to do that and thus we can avoid a redundant Mutex.